lib/build/browserify.js

```diff
@@ -179,43 +179,52 @@ function compile(bannerWidth, transforms) {
-        message = [analysis[3], analysis[1], 0, ' ' + analysis[2]].join(':') + '\n';
-
-      // Error: Cannot find module '<reason>' from '<directory>'
-      //  find the first text match for any text quoted in <reason>
-      } else if (analysis = /^\s*Error\: Cannot find module '(.*)\'\s*from\s*\'(.*)\'\s*$/.exec(text)) {
-        var filename = fs.readdirSync(analysis[2])
-          .filter(RegExp.prototype.test.bind(/\.js$/i))
-          .filter(function (jsFilename) {
-            var fullPath = path.join(analysis[2], jsFilename);
-            var fileText = fs.readFileSync(fullPath).toString();
-            return (new RegExp('[\'"]' + analysis[1] + '[\'"]')).test(fileText);
-          })
-          .shift();
-        message = path.join(analysis[2], filename) + ':0:0: Cannot find import ' + analysis[1] + '\n';
-
-      // Unknown
-      } else {
-        message = 'TODO parse this error\n' + text + '\n';
-      }
-
-      // add unique
-      if (output.indexOf(message) < 0) {
-        output.push(message);
-      }
+      [
+        // SyntaxError: <file>:<reason>:<line>:<column>
+        function() {
+          var analysis = /^\s*SyntaxError\:\s*([^:]*)\s*\:\s*([^(]*)\s*\((\d+:\d+)\)\s*\n/.exec(text);
+          return analysis && ([analysis[1], analysis[3], analysis[2]].join(':') + '\n');
+        },
+        // Error: SyntaxError: <reason> while parsing json file <file>
+        function() {
+          var analysis = /^\s*Error: SyntaxError\:\s*(.*)\s*while parsing json file\s*([^]*)/.exec(text);
+          return analysis && ([analysis[2], '0', '0', ' ' + analysis[1]].join(':') + '\n');
+        },
+        // Line <line>: <reason>: <file>
+        function() {
+          var analysis = /Line\s*(\d+)\s*\:\s*([^:]*)\s*:\s*(.*)\s*/.exec(text);
+          return analysis && ([analysis[3], analysis[1], 0, ' ' + analysis[2]].join(':') + '\n');
+        },
+        // Error: Cannot find module '<reason>' from '<directory>'
+        //  find the first text match for any text quoted in <reason>
+        function() {
+          var analysis = /^\s*Error\: Cannot find module '(.*)\'\s*from\s*\'(.*)\'\s*$/.exec(text);
+          if (analysis) {
+            var filename = fs.readdirSync(analysis[2])
+              .filter(RegExp.prototype.test.bind(/\.js$/i))
+              .filter(function (jsFilename) {
+                var fullPath = path.join(analysis[2], jsFilename);
+                var fileText = fs.readFileSync(fullPath).toString();
+                return (new RegExp('[\'"]' + analysis[1] + '[\'"]')).test(fileText);
+              })
+              .shift();
+            return path.join(analysis[2], filename) + ':0:0: Cannot find import ' + analysis[1] + '\n';
+          }
+        },
+        // Unknown
+        function() {
+          return 'TODO parse this error\n' + text + '\n';
+        }
+      ]
+        .map(function(method) {
+          return method();
+        })
+        .filter(Boolean)
```

Comment:
This is not as intended. We want the final method above to be the fall-through. As it stands, it will be added always.